src/constants.rs

```diff
@@ -35,6 +35,9 @@ pub const A: FieldElement       = FieldElement([
     486662, 0, 0, 0, 0, 0, 0, 0, 0, 0, ]);
 
 /// SQRT_MINUS_A is sqrt(-486662)
+// XXX I think that this was used in Adam's code for his elligator
+// implementation, but that should maybe be using sqrt(-486664)
+// instead...?  - hdevalence
```

Comment:
This is the part of the forward mapping where one checks that `(s² - 2)(s² + 2) ≠ 0` where `s ∈ 𝐆𝐅(q)`? I think I'm not seeing where the `-A - 2` comes in.